pkg/subctl/cmd/join.go

```diff
@@ -277,16 +278,31 @@ func joinSubmarinerCluster(config clientcmd.ClientConfig, contextName string, su
 	status.End(cli.CheckForError(err))
 	exitOnError("Error creating SA for cluster", err)
 
-	status.Start("Deploying Submariner")
-	err = submarinercr.Ensure(clientConfig, OperatorNamespace, populateSubmarinerSpec(subctlData, netconfig))
-	if err == nil {
-		status.QueueSuccessMessage("Submariner is up and running")
-		status.End(cli.Success)
-	} else {
-		status.QueueFailureMessage("Submariner deployment failed")
-		status.End(cli.Failure)
+	if subctlData.IsConnectivityEnabled() {
+		status.Start("Deploying Submariner")
+		err = submarinercr.Ensure(clientConfig, OperatorNamespace, populateSubmarinerSpec(subctlData, netconfig))
+		if err == nil {
+			status.QueueSuccessMessage("Submariner is up and running")
+			status.End(cli.Success)
+		} else {
+			status.QueueFailureMessage("Submariner deployment failed")
+			status.End(cli.Failure)
+		}
+
+		exitOnError("Error deploying Submariner", err)
+	} else if subctlData.IsServiceDiscoveryEnabled() {
```

Comment:
How about adding a `ConnectivityEnabled` flag to the `Submariner` CR so we can let the operator controller handle this normally?

Reply:
It's the operator controller itself :)  we designed it that way. This is why we split it in two objects "Submariner "  & "ServiceDiscovery"

But in most cases submariner will bring service discovery, so we generate that object automatically. But we wanted to allow ServiceDiscovery being created alone, but never added support to subctl.

Reply:
The `SubmarinerReconciler` in the operator controller already creates the `ServiceDiscovery` CR based on the `ServiceDiscoveryEnabled` flag in the `Submariner` CR.  It seems to make sense to keep that workflow.

Reply:
But then it doesn't make sense to keep a separate API on the submariner-operator controller. It could just be a sub-package handling those details.

It's the current design, not written in stone anyway, but if we do what you suggest I would kill the ServiceDiscovery CR altogether. @vthapar @skitt   ^

I lean more towards keeping ServiceDiscover TBH.

Reply:
I think we should do it one way or the other.  I lean towards removing the `ServiceDiscoveryEnabled` flag altogether and having `subctl` create the `ServiceDiscovery` CR as you've done here. We can do the flag removal part in another PR.

Reply:
Right, "ServiceDiscoveryEnabled" flag exists so the people using the bare operator doesn't need to create 2 objects. I think it would be fine if we move all those pesky credentials to a kubernetes secret instead, so the broker credentials live in a single place.

@vthapar thoughts ^